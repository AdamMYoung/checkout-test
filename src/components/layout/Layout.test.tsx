import { render, screen, within } from '@testing-library/react';
import { axe } from 'jest-axe';
import { Layout } from '.';

describe('Layout', () => {
    it('passes aXe validation', async () => {
        const { container } = render(<Layout>Test</Layout>);

        expect(await axe(container)).toHaveNoViolations();
    });

    it('matches snapshot', () => {
        const { container } = render(<Layout>Test</Layout>);

        expect(container).toMatchSnapshot();
    });

    it('renders as main, with children inside of main', () => {
        const children = 'Hello World!';
        render(<Layout>{children}</Layout>);

        const main = screen.getByRole('main');
        const text = within(main).getByText(children);

        expect(main).toBeInTheDocument();
        expect(text).toBeInTheDocument();
    });
});
