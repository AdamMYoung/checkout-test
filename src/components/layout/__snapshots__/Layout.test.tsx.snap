// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<div>
  <div
    class="css-0"
  >
    <main
      class="chakra-container css-7nseej"
    >
      Test
    </main>
  </div>
</div>
`;
